pkg/pipelines/argocd/argocd.go

```diff
@@ -107,6 +149,28 @@ func makeSource(env *config.Environment, app *config.Application, repoURL string
 	}
 }
 
+func ignoreDifferences(app *argoappv1.Application) *argoappv1.Application {
+	app.Spec.IgnoreDifferences = ignoreDifferencesFields
+	return app
+}
+
+func argoCDResource(ns string) (argov1.ArgoCD, error) {
```

Comment:
The `resourceExclusions` field in the ArgoCD resource expects a string. I added a custom struct that represents that field and converted it to a string. The `resource` struct is not from upstream. Should I return nil in case of an error?